arch/Mips/MipsMapping.c

```diff
@@ -1,1071 +1,467 @@
+	}
+}
 
-	{ MIPS_INS_JALR_HB, "jalr.hb" },
-	{ MIPS_INS_JR_HB, "jr.hb" },
+static const map_insn_ops insn_operands[] = {
+#include "MipsGenCSMappingInsnOp.inc"
 };
 
-const char *Mips_insn_name(csh handle, unsigned int id)
+static void Mips_set_detail_op_mem_reg(MCInst *MI, unsigned OpNum, mips_reg Reg)
 {
-#ifndef CAPSTONE_DIET
-	if (id >= MIPS_INS_ENDING)
-		return NULL;
+	Mips_get_detail_op(MI, 0)->type = MIPS_OP_MEM;
+	Mips_get_detail_op(MI, 0)->mem.base = Reg;
+	Mips_get_detail_op(MI, 0)->access = map_get_op_access(MI, OpNum);
+}
 
-	return insn_name_maps[id].name;
-#else
-	return NULL;
-#endif
+static void Mips_set_detail_op_mem_disp(MCInst *MI, unsigned OpNum, int64_t Imm)
+{
+	Mips_get_detail_op(MI, 0)->type = MIPS_OP_MEM;
+	Mips_get_detail_op(MI, 0)->mem.disp = Imm;
+	Mips_get_detail_op(MI, 0)->access = map_get_op_access(MI, OpNum);
 }
 
-#ifndef CAPSTONE_DIET
-static const name_map group_name_maps[] = {
-	// generic groups
-	{ MIPS_GRP_INVALID, NULL },
-	{ MIPS_GRP_JUMP, "jump" },
-	{ MIPS_GRP_CALL, "call" },
-	{ MIPS_GRP_RET, "ret" },
-	{ MIPS_GRP_INT, "int" },
-	{ MIPS_GRP_IRET, "iret" },
-	{ MIPS_GRP_PRIVILEGE, "privileged" },
-	{ MIPS_GRP_BRANCH_RELATIVE, "branch_relative" },
+static void Mips_set_detail_op_imm(MCInst *MI, unsigned OpNum, int64_t Imm)
+{
+	if (!detail_is_set(MI))
+		return;
 
-	// architecture-specific groups
-	{ MIPS_GRP_BITCOUNT, "bitcount" },
-	{ MIPS_GRP_DSP, "dsp" },
-	{ MIPS_GRP_DSPR2, "dspr2" },
-	{ MIPS_GRP_FPIDX, "fpidx" },
-	{ MIPS_GRP_MSA, "msa" },
-	{ MIPS_GRP_MIPS32R2, "mips32r2" },
-	{ MIPS_GRP_MIPS64, "mips64" },
-	{ MIPS_GRP_MIPS64R2, "mips64r2" },
-	{ MIPS_GRP_SEINREG, "seinreg" },
-	{ MIPS_GRP_STDENC, "stdenc" },
-	{ MIPS_GRP_SWAP, "swap" },
-	{ MIPS_GRP_MICROMIPS, "micromips" },
-	{ MIPS_GRP_MIPS16MODE, "mips16mode" },
-	{ MIPS_GRP_FP64BIT, "fp64bit" },
-	{ MIPS_GRP_NONANSFPMATH, "nonansfpmath" },
-	{ MIPS_GRP_NOTFP64BIT, "notfp64bit" },
-	{ MIPS_GRP_NOTINMICROMIPS, "notinmicromips" },
-	{ MIPS_GRP_NOTNACL, "notnacl" },
+	if (doing_mem(MI)) {
+		Mips_set_detail_op_mem_disp(MI, OpNum, Imm);
+		return;
+	}
+
+	Mips_get_detail_op(MI, 0)->type = MIPS_OP_IMM;
+	Mips_get_detail_op(MI, 0)->imm = Imm;
+	Mips_get_detail_op(MI, 0)->access = map_get_op_access(MI, OpNum);
+	Mips_inc_op_count(MI);
+}
 
-	{ MIPS_GRP_NOTMIPS32R6, "notmips32r6" },
-	{ MIPS_GRP_NOTMIPS64R6, "notmips64r6" },
-	{ MIPS_GRP_CNMIPS, "cnmips" },
+static void Mips_set_detail_op_uimm(MCInst *MI, unsigned OpNum, uint64_t Imm)
+{
+	if (!detail_is_set(MI))
+		return;
 
-	{ MIPS_GRP_MIPS32, "mips32" },
-	{ MIPS_GRP_MIPS32R6, "mips32r6" },
-	{ MIPS_GRP_MIPS64R6, "mips64r6" },
+	if (doing_mem(MI)) {
+		Mips_set_detail_op_mem_disp(MI, OpNum, Imm);
+		return;
+	}
 
-	{ MIPS_GRP_MIPS2, "mips2" },
-	{ MIPS_GRP_MIPS3, "mips3" },
-	{ MIPS_GRP_MIPS3_32, "mips3_32"},
-	{ MIPS_GRP_MIPS3_32R2, "mips3_32r2" },
+	Mips_get_detail_op(MI, 0)->type = MIPS_OP_IMM;
+	Mips_get_detail_op(MI, 0)->imm = (int64_t)Imm;
+	Mips_get_detail_op(MI, 0)->is_unsigned = true;
```

Comment:
As said above. If `uint64` values exist, you have the choice between changing the API here (and make it a `uint64_t`) or document it in code and the release document. In any case it just needs to be clear who the field should be interpreted. By bits or by type.

When you refactor Rizin and see that one interpretation is vastly better, you can still change it. Preferable though before Alpha and at latest before the Beta release.